syz-manager/manager.go

```diff
@@ -1066,7 +1055,114 @@ func (mgr *Manager) fuzzerConnect() ([]rpctype.RPCInput, BugFrames, bool) {
 	for frame := range mgr.dataRaceFrames {
 		dataRaceFrames = append(dataRaceFrames, frame)
 	}
-	return corpus, BugFrames{memoryLeaks: memoryLeakFrames, dataRaces: dataRaceFrames}, mgr.coverFilterFilename != ""
+	return corpus, BugFrames{memoryLeaks: memoryLeakFrames, dataRaces: dataRaceFrames}
+}
+
+const EXT string = ".ko"
+
+func findModules(dir string) ([]string, error) {
+	files := []string{}
+	err := filepath.Walk(dir, func(path string, f os.FileInfo, err error) error {
+		if filepath.Ext(path) == EXT {
+			files = append(files, path)
+		}
+		return nil
+	})
+
+	return files, err
+}
+
+func (mgr *Manager) getModulePath(name string) (string, error) {
+	modulePath := ""
+	if mgr.cfg.KernelObj != "" {
+		paths, err := findModules(mgr.cfg.KernelObj)
+		if err != nil {
+			return modulePath, err
+		}
+		for _, path := range paths {
+			if strings.TrimSuffix(filepath.Base(path), EXT) == name {
+				return path, nil
+			}
+		}
+	}
+	if mgr.cfg.ModuleObj != "" {
+		paths, err := findModules(mgr.cfg.ModuleObj)
+		if err != nil {
+			return modulePath, err
+		}
+		for _, path := range paths {
+			if strings.TrimSuffix(filepath.Base(path), EXT) == name {
+				return path, nil
+			}
+		}
+	}
+
+	return modulePath, fmt.Errorf("no match module path found")
+}
+
+func (mgr *Manager) updateKernelModules(a *rpctype.ConnectArgs) error {
+	count := 0
+	for _, rmodule := range a.Modules {
+		hasKey := false
+		i := 0
+		for idx, module := range mgr.cfg.KernelModules {
+			if module.Name == rmodule.Name {
+				hasKey = true
+				i = idx
+				break
+			}
+		}
+		if hasKey {
+			if mgr.cfg.KernelModules[i].Path == "" && rmodule.Name != "" {
+				path, err := mgr.getModulePath(rmodule.Name)
+				if err != nil && !mgr.cfg.KernelModules[i].Disabled {
+					log.Logf(0, "module path needs to be specified in config for %v", rmodule.Name)
+					return fmt.Errorf("module path needs to be specified in config for %v", rmodule.Name)
+				}
+				mgr.cfg.KernelModules[i].Path = path
+			}
+			mgr.cfg.KernelModules[i].Addr = rmodule.Addr
+		} else {
+			if rmodule.Name == "" {
+				mgr.cfg.KernelModules = append(mgr.cfg.KernelModules, backend.KernelModule{
+					Name: "",
+					Addr: rmodule.Addr,
+				})
+				continue
+			}
+			path, err := mgr.getModulePath(rmodule.Name)
+			if err != nil {
+				log.Logf(0, "module name and path needs to be specified in config for %v", rmodule.Name)
+				count++
+			}
+			mgr.cfg.KernelModules = append(mgr.cfg.KernelModules, backend.KernelModule{
+				Name: rmodule.Name,
+				Addr: rmodule.Addr,
+				Path: path,
+			})
+		}
+	}
+	if count != 0 {
+		return fmt.Errorf("failed to auto find module path")
+	}
+	log.Logf(0, "kernel modules: %v", mgr.cfg.KernelModules)
```

Comment:
If mgr.httpCoverCover is called before this (user tries to open coverage report before first VM connect), report generator will be initialized without modules (or without addresses).
I think httpCoverCover need to check if modules are initialized (or that machine is checked, or we have any coverage) and return early without calling getReportGenerator.